reference/forms/types/choice.rst

```diff
@@ -229,9 +298,14 @@ choice_loader
 
 **type**: :class:`Symfony\\Component\\Form\\ChoiceList\\Loader\\ChoiceLoaderInterface`
 
-The ``choice_loader`` can be used to only partially load the choices in cases where
-a fully-loaded list is not necessary. This is only needed in advanced cases and
-would replace the ``choices`` option.
+The ``choice_loader`` can be used to load the choices form a data source with a
+custom logic (e.g query language) such as database or search engine.
+The list will be fully loaded to display the form, but while submission only the
+submitted choices will be loaded.
```

Comment:
... display the form, but on submission, only the submitted choices will be loaded.